pkg/apis/resource/validation/validation_resourceclaim_test.go

```diff
@@ -359,18 +370,58 @@ func TestValidateClaimStatusUpdate(t *testing.T) {
 			update: func(claim *resource.ResourceClaim) *resource.ResourceClaim {
 				claim.Status.DriverName = "valid"
 				claim.Status.Allocation = &resource.AllocationResult{
-					ResourceHandle: strings.Repeat(" ", resource.ResourceHandleMaxSize),
+					Context: []resource.ResourceHandle{
+						{
+							Data: strings.Repeat(" ", resource.ResourceHandleDataMaxSize),
+						},
+					},
+				}
+				return claim
+			},
+		},
+		"invalid-allocation-context": {
+			wantFailures: field.ErrorList{field.TooLongMaxLength(field.NewPath("status", "allocation", "context"), resource.AllocationResultContextMaxSize+1, resource.AllocationResultContextMaxSize)},
+			oldClaim:     validClaim,
+			update: func(claim *resource.ResourceClaim) *resource.ResourceClaim {
+				claim.Status.DriverName = "valid"
+				claim.Status.Allocation = &resource.AllocationResult{
+					Context: func() []resource.ResourceHandle {
+						var handles []resource.ResourceHandle
+						for i := 0; i < resource.AllocationResultContextMaxSize+1; i++ {
```

Comment:
nit: "i <= resource.AllocationResultContextMaxSize" ?

Reply:
I adapted this from what you had previously, and actually think the +1 is clearer since the goal is to check an out of bounds error. I’d rather leave it as is unless you feel strongly.

Reply:
As I am not consistent with what I prefer, let's leave it as-is :sweat: